src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -104,26 +104,27 @@ g.test('stale')
   )
   .paramsSubcasesOnly(u =>
     u
-      .combine('initialError', [undefined, 'TypeError', 'OperationError'])
-      .combine('awaitInitialError', [true, false])
-      .combine('awaitSuccess', [true, false])
-      .unless(
-        ({ initialError, awaitInitialError }) => initialError === undefined && awaitInitialError
-      )
+      .combine('request1', ['None', 'Valid', 'TypeError', 'OperationError'] as const)
+      .expand('awaitRequest1', p => (p.request1 === 'None' ? [false] : [true, false]))
+      .combine('awaitRequest2', [true, false])
   )
   .fn(async t => {
     const gpu = getGPU(t.rec);
     const adapter = await gpu.requestAdapter();
     assert(adapter !== null);
 
-    const { initialError, awaitInitialError, awaitSuccess } = t.params;
+    const { request1, awaitRequest1, awaitRequest2 } = t.params;
 
-    switch (initialError) {
-      case undefined:
+    switch (request1) {
+      case 'None':
+        // Cause an operation error by requesting a device on a consumed adapter.
+        t.shouldReject('OperationError', t.requestDeviceTracked(adapter));
+        break;
+      case 'Valid':
         break;
```

Comment:
There haven't been any requestDevice calls yet, why would it be consumed?

What I was thinking here was:
```suggestion
      case 'None':
        break;
      case 'Valid':
        if (awaitRequest1) {
          await t.requestDeviceTracked(adapter);
        } else {
          t.requestDeviceTracked(adapter);
        }
        break;
```

And then later on, request2 is where you check for an OperationError for a consumed adapter.

Reply:
No request2 parameter, but there is a second requestDevice call. That's the one that we're testing. The first requestDevice is just setup for the second one.

Reply:
Gotcha! I've updated code.